samples/coop_synchronization/README.rst

```diff
@@ -0,0 +1,71 @@
+.. _coop_synchronization_sample:
+
+Cooperative Synchronization Sample
+##################################
+
+Overview
+********
+
+A simple application that demonstrates even more basic sanity of the kernel.
+Two threads (A and B) take turns printing a greeting message to the console.
+They use yields and semaphores to control the whose turn it is to print.
+The rate is only controlled via busy waits. This demonstrates kernel communication only.
+This is a good example to test new platforms that may not have a working systick.
```

Comment:
Terminology nitpick: "SysTick" is the name of the ARM Cortex M timer hardware (c.f. our cortex_m_systick.c timer driver).  The Zephyr term for the subsystem that announces ticks is just "clock" or "timer".  So for clarity it's probably best to explain that this works when CONFIG_SYS_CLOCK_EXISTS=n.

Reply:
I think that's a great idea, when I try the linking fails with an undefined reference to `smp_timer_init()`, which is not that surprising. 
I can't decide which direction I should go from here; filter out `smp` enabled boards OR not introduce `CONFIG_SYS_CLOCK_EXISTS=n`. 
May I ask for your opinion on this?